gcloud-java-dns/src/test/java/com/google/gcloud/dns/ITDnsTest.java

```diff
@@ -0,0 +1,1268 @@
+  public static final String ZONE_DESCRIPTION1 = "first zone";
+  public static final String ZONE_DNS_NAME1 = ZONE_NAME1 + ".com.";
+  public static final String ZONE_DNS_EMPTY_DESCRIPTION = ZONE_NAME_EMPTY_DESCRIPTION + ".com.";
+  public static final String ZONE_DNS_NAME_NO_PERIOD = ZONE_NAME1 + ".com";
+  public static final ZoneInfo ZONE1 = ZoneInfo.builder(ZONE_NAME1)
+      .description(ZONE_DESCRIPTION1)
+      .dnsName(ZONE_DNS_EMPTY_DESCRIPTION)
+      .build();
+  public static final ZoneInfo ZONE_EMPTY_DESCRIPTION =
+      ZoneInfo.builder(ZONE_NAME_EMPTY_DESCRIPTION)
+          .description(ZONE_DESCRIPTION1)
+          .dnsName(ZONE_DNS_NAME1)
+          .build();
+  public static final ZoneInfo ZONE_NAME_ERROR =
+      ZoneInfo.builder(ZONE_NAME_TOO_LONG)
+          .description(ZONE_DESCRIPTION1)
+          .dnsName(ZONE_DNS_NAME1)
+          .build();
+  public static final ZoneInfo ZONE_MISSING_DESCRIPTION =
+      ZoneInfo.builder(ZONE_NAME1)
+          .dnsName(ZONE_DNS_NAME1)
+          .build();
+  public static final ZoneInfo ZONE_MISSING_DNS_NAME =
+      ZoneInfo.builder(ZONE_NAME1)
+          .description(ZONE_DESCRIPTION1)
+          .build();
+  public static final ZoneInfo ZONE_DNS_NO_PERIOD =
+      ZoneInfo.builder(ZONE_NAME1)
+          .description(ZONE_DESCRIPTION1)
+          .dnsName(ZONE_DNS_NAME_NO_PERIOD)
+          .build();
+  public static final DnsRecord A_RECORD_ZONE1 = DnsRecord
+      .builder("www." + ZONE1.dnsName(), DnsRecord.Type.A)
+      .records(ImmutableList.of("123.123.55.1"))
+      .ttl(25, TimeUnit.SECONDS)
+      .build();
+  public static final DnsRecord AAAA_RECORD_ZONE1 = DnsRecord
+      .builder("www." + ZONE1.dnsName(), DnsRecord.Type.AAAA)
+      .records(ImmutableList.of("ed:ed:12:aa:36:3:3:105"))
+      .ttl(25, TimeUnit.SECONDS)
+      .build();
+  public static final ChangeRequest CHANGE_ADD_ZONE1 = ChangeRequest.builder()
+      .add(A_RECORD_ZONE1)
+      .add(AAAA_RECORD_ZONE1)
+      .build();
+  public static final ChangeRequest CHANGE_DELETE_ZONE1 = ChangeRequest.builder()
+      .delete(A_RECORD_ZONE1)
+      .delete(AAAA_RECORD_ZONE1)
+      .build();
+
+  public static void purge() {
+    Page<Zone> zones = DNS.listZones();
+    Iterator<Zone> zoneIterator = zones.iterateAll();
+    while (zoneIterator.hasNext()) {
+      Zone zone = zoneIterator.next();
+      List<DnsRecord> toDelete = new LinkedList<>();
+      if (zone.name().startsWith(PREFIX)) {
+        Iterator<DnsRecord> dnsRecordIterator = zone.listDnsRecords().iterateAll();
+        while (dnsRecordIterator.hasNext()) {
+          DnsRecord record = dnsRecordIterator.next();
+          if (!ImmutableList.of(DnsRecord.Type.NS, DnsRecord.Type.SOA).contains(record.type())) {
+            toDelete.add(record);
+          }
+        }
+        if (!toDelete.isEmpty()) {
+          zone.applyChangeRequest(ChangeRequest.builder().deletions(toDelete).build());
+        }
+        zone.delete();
```

Comment:
This is probably working because the time the second request is issued the change request has already been processed. However we don't have guarantees on that, so that could lead to unexpected failures. I suggest we wait for change to complete.